destreamer.ts

```diff
@@ -77,34 +77,28 @@ function init() {
         console.info(colors.blue("There will be no video downloaded, it's only a simulation\n"));
 }
 
-async function DoInteractiveLogin(username?: string): Promise<Session> {
+async function DoInteractiveLogin(url: string, username?: string): Promise<Session> {
+
+    let videoID = url.split("/").pop() ?? process.exit(25);
+
     console.log('Launching headless Chrome to perform the OpenID Connect dance...');
     const browser = await puppeteer.launch({
         headless: false,
         args: ['--disable-dev-shm-usage']
     });
     const page = (await browser.pages())[0];
-    console.log('Navigating to microsoftonline.com login page...');
+    console.log('Navigating to login page...');
 
-    // This breaks on slow connections, needs more reliable logic
-    await page.goto('https://web.microsoftstream.com', { waitUntil: 'networkidle2' });
+    await page.goto(url, { waitUntil: 'load' });
     await page.waitForSelector('input[type="email"]');
 
     if (username) {
         await page.keyboard.type(username);
         await page.click('input[type="submit"]');
     }
 
-    await browser.waitForTarget(target => target.url().includes('microsoftstream.com/'), { timeout: 90000 });
+    await browser.waitForTarget(target => target.url().includes(videoID), { timeout: 150000 });

```

Comment:
I like it! includes(videoID) is probably the right thing here.

Reply:
sure can do, fixing it